core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -634,6 +636,10 @@ boolean isPreserveAliases() {
                 if (!noIndicesAllowedOrMatches(options, matches)) {
                     throw infe(expression);
                 }
+
+                if (expression.contains("*")) {
```

Comment:
> Should `-test1,test*` resolve to the same indices as `test*,-test1`? It doesn't now right?

It doesn't right now, I think it would depend on if a `-test1` index existed.

I don't think it should though, since the intended behavior would be different regardless if you swapped the ordering of negations? What do you think?

Maybe instead we should start with a list of indices and work backwards towards what we expect for certain combinations?